examples/query.rs

```diff
@@ -0,0 +1,40 @@
+use serde::{Deserialize, Serialize};
+use surrealdb_rs::param::Root;
+use surrealdb_rs::protocol::Ws;
+use surrealdb_rs::{Result, Surreal};
+
+#[derive(Debug, Deserialize)]
+#[allow(dead_code)]
+struct User {
+    id: String,
+    name: String,
+}
+
+#[derive(Debug, Serialize)]
+struct Params<'a> {
+    name: &'a str,
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let mut conn = Surreal::connect(Ws("localhost:8000")).await?;
+
+    conn.signin(Root {
+        user: "root",
+        pass: "root",
+    })
+    .await?;
+
+    conn.use_ns("namespace").use_db("database").await?;
+
+    let _users: Vec<User> = conn.query("SELECT * FROM user").await?;
```

Comment:
`query` can take multiple statements, so what happens if there is a second statement?

Reply:
> `query` can take multiple statements, so what happens if there is a second statement?

You mean in the response? So far, the idea was to leave handling the response to the user as they know best what their queries will return. If they will return multiple types, then instead of requesting a particular type, they can accept a `Value`

```rust
let _values: Vec<Value> = conn.query("CREATE foo; SELECT * FROM user;").await?;
```

and handle deserialising those values themselves. What did you have in mind on how we can tackle this?